cs-couchbase/src/main/java/io/cloudslang/content/couchbase/factory/cluster/ClusterUriFactory.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * (c) Copyright 2017 Hewlett-Packard Development Company, L.P.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License v2.0 which accompany this distribution.
+ *
+ * The Apache License is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+*/
+/*******************************************************************************
+ * (c) Copyright 2017 Hewlett-Packard Development Company, L.P.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License v2.0 which accompany this distribution.
+ *
+ * The Apache License is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *******************************************************************************/
+package io.cloudslang.content.couchbase.factory.cluster;
+
+import io.cloudslang.content.couchbase.entities.inputs.InputsWrapper;
+
+import static io.cloudslang.content.couchbase.entities.constants.Constants.ClusterActions.GET_CLUSTER_DETAILS;
+import static io.cloudslang.content.couchbase.entities.constants.Constants.ClusterActions.GET_CLUSTER_INFO;
+import static io.cloudslang.content.couchbase.entities.constants.Constants.ClusterActions.GET_DESTINATION_CLUSTER_REFERENCE;
+import static io.cloudslang.content.couchbase.entities.constants.Constants.ClusterActions.REBALANCING_NODES;
+import static io.cloudslang.content.couchbase.entities.couchbase.CouchbaseApi.CONTROLLER;
+import static io.cloudslang.content.couchbase.entities.couchbase.CouchbaseApi.POOLS;
+import static io.cloudslang.content.couchbase.entities.couchbase.CouchbaseApi.POOLS_DEFAULT;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+
+public class ClusterUriFactory {
+    private ClusterUriFactory() {
+        // prevent instantiation
+    }
+
+    public static String getClusterUriValue(InputsWrapper wrapper) {
+        String action = wrapper.getCommonInputs().getAction();
```

Comment:
is this nullable? In that case maybe you should check

Reply:
wrapper object is not null... it contains at least http client minimal settings (defaults) and action string is set through @Action without user interference